### DISCUSSION TITLE
whether there is a Normalization operator that can fix my total power at a certain value

### DISCUSSION THREAD

**Discussion by js-jixu** [idaholab/moose#23068] [Q&A General] [answered]
I want to obtain power data from an exodus file through SolutionUserObject for another simulation of thermal expansion. Since the power is a MONOMIAL variable, when the volume of an element becomes larger, the power represented by this element will also become larger. So I want to know whether there is a Normalization operator that can fix my total power at a certain value

**Comment by GiudGiud** [comment]
There s a NormalizationAux you can use. A parsedAux will also work with a postprocessor computing the norm

**Reply by js-jixu** [reply]
Can one AuxVariable use two AuxKernels? Because `power` needs to use SolutionAux to obtain the power distribution in the exodus file. 

```
[AuxVariables]
  [power]
    family = MONOMIAL
    order = FIRST
  []
[]

[AuxKernels]
  [power_auxk]
    type = SolutionAux
    variable = power
    solution = power_uo
  []
[]

[UserObjects]
  [power_uo]
    type = SolutionUserObject
    mesh = initial.e
    system_variables = power
    timestep = LATEST
    execute_on = INITIAL
  []
[]
```

In this case, can `power` use `NormalizationAux` for normalization?

**Reply by GiudGiud** [reply]
Just use two variables to be safe

**Reply by GiudGiud** [reply]
But yeah we can on paper we just have to be careful to order them properly so it might be disallowed

**Comment by hugary1995** [comment, marked as answer]
1. It's okay to have multiple AuxKernels operating on the same AuxVariable as long as they do not overlap in space (subdomains) or in time (execute_on).
2. I'd think carefully about this statement "Since the power is a MONOMIAL variable, when the volume of an element becomes larger, the power represented by this element will also become larger". The power density is a volumetric quantity, it is independent of the element size unless you integrate it in space.

**Reply by GiudGiud** [reply]
He's in the right for 2. If you keep the same power density in an element but grow the element you are increasing power for no good reason.

**Reply by GiudGiud** [reply]
Actually using a full domain renormalization might not be the best way to do this.
I think we would rather normalize element by element, to make sure the power in each element stays the same
